libretroshare/src/services/p3msgservice.cc

```diff
@@ -1988,13 +1997,38 @@ void p3MsgService::manageDistantPeers()
             std::cerr << (mShouldEnableDistantMessaging?"Enabling":"Disabling") << " distant messaging, with peer id = " << *it << std::endl;
 #endif
 
-        for(std::list<RsGxsId>::const_iterator it(own_id_list.begin());it!=own_id_list.end();++it)
-        {
-            if(mShouldEnableDistantMessaging)
-                mGRouter->registerKey(*it,GROUTER_CLIENT_ID_MESSAGES,"Messaging contact") ;
-            else
-                mGRouter->unregisterKey(*it,GROUTER_CLIENT_ID_MESSAGES) ;
-        }
+		for( auto it = std::as_const(own_id_list).begin();
+		     it != own_id_list.end(); ++it )
+		{
+			if(mShouldEnableDistantMessaging)
+			{
+#ifndef V07_NON_BACKWARD_COMPATIBLE_CHANGE_GROUTER_SERVICE_ID
+				// Keep for RetroShare < 0.6.6 retro-compatibility
+				mGRouter->registerKey(
```

Comment:
Same here! You get to register 2 GRouter clients in order to keep backward compat!